Yoast/wordpress-seo
# Add twitter title presenter

## Summary


This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Implemented `twitter_title`, including unit tests
* Also added unit tests for `title`, since they were missing and `twitter_title` is based on this
* Fixed CS and documentation in unrelated files when I came across problems

## Test instructions

This PR can be tested by following these steps:

* The page source should show an indented `twitter-title` tag with the correct title. This is the example for a simple post:
<img width="695" alt="Screenshot 2019-10-01 at 15 02 07" src="https://user-images.githubusercontent.com/20280513/65964144-8fab4080-e45c-11e9-922a-452882d026f4.png">

* When testing, (also) use snippet variables in your `twitter-title`, for example `%%date%%`. Make sure they are output correctly (this example should be output as a formatted date, for example `"September 26, 2019"`).

* You should check whether the `twitter-title` tag is present for the following types of content
    * Post 
        * Create a post (`Posts` --> `Add new` --> `Social` (in metabox) --> add a `Twitter title`)
        * `Publish` --> `View post`
        * Right click --> `View page source` (this bullet point also applies to all examples below, and will be left out from now on)
    * Page
        * Create a page (`Pages` --> `Add new` --> `Social` (in metabox) --> add a `Twitter title`)
        * `Publish` --> `View page`
    * Attachment page
        * STATUS :heavy_multiplication_x:: `twitter-title` tag is shown, but contents are incorrect
        * In the menu on the left, click `SEO` --> `Search appearance` --> `Media`
        * Set `Redirect attachment URLs to the attachment itself?` to `No`, set `Yoast SEO Meta Box` to `Show`, then click `Save changes`
        * In the menu on the left, click `Media`
        * Click on an image (or first add an image if the media library is empty), then click `View attachment page`
        * In the top menu, click `Edit media`. This will show the metabox, click `Social` --> add a `Twitter title` --> `Update`
        * View the attachment page again (it seems that for this, you need to follow the first two above steps again)
    * Another public and publicly queryable post type
        * Activate the `Yoast Test Helper` plugin; `Books` and `Movies` will now appear in the menu on the left.
        * `Books` --> `Add new` --> `Social` (in metabox) --> add a `Twitter title`
        * `Publish` --> `View post`
    * Post type archive for posts
        * In the menu on the left, click `Settings` --> `Reading`
        * Set `Your homepage displays` to `A static page`
        * For `Posts page`, set a random page, then `Save changes`
        * Through the menu on the left, navigate to the page you just set as homepage, `Edit` --> `Social` (in metabox) --> add a `Twitter title`
        * `Update` and `View page`
        * This page now represents a post type archive for posts
    * Post type archive for another public and publicly queryable post type
        * View a book post (see above how to create such a post)
        * In the bread crumb, click `Book` to go to the archive
        * `twitter-title` should be output in the page source, but its contents default to the general `title`, since there is no way to set `twitter-title`. 
    * Term archive for categories
        * Go to `Posts`, and add categories to a few posts (to add a category, create/edit a post; in the right menu there is a `Categories` collapsible)
        * In the left menu, click `Posts` --> `Categories`
        * Hoover over a category name, then click `Edit` and set a Twitter title
        * Go back, hoover over the category name again, then click `View`
    * Term archive for tags
        * Exactly the same as for `Categories`, but for `Tags`
    * Term archive for another public and publicly queryable post type
        * Hoover over `Books` in the left menu, there you see `Categories` and `Genres` (these genres correspond to `Tags`)
        * Follow the steps described above for `Categories` and `Tags`
    * Author archives
        * STATUS :heavy_multiplication_x:: `twitter-title` is not shown
        * Navigate to an author archive, for example `http://one.wordpress.test/author/admin/` (potentially replace `one` with the name of your installation`)
    * Date archives
        * STATUS :heavy_multiplication_x:: `twitter-title` is not shown
        * Navigate to a date archive, for example `http://one.wordpress.test/2019/09/13/` 
    * Home page
        * Navigate to the homepage
    * Error page
        * STATUS :heavy_multiplication_x:: `twitter-title` is not shown
        * Navigate to a page that can't be found, for example `http://one.wordpress.test/aaa`
    * Search result page
        * STATUS :heavy_multiplication_x:: `twitter-title` is not shown
        * Navigate to a search result page, for example `http://one.wordpress.test/?s=gutenberg`
    * Site wide default
        * STATUS ❓ : how to navigate to a site wide default?

## UI changes

## Documentation

## Quality assurance


Fixes #
